install_config/certificate_customization.adoc

```diff
@@ -19,82 +19,277 @@ This can be done during an
+
+----
+openshift_hosted_router_certificate={"certfile": "/path/to/apps.c1-ocp.myorg.com.crt", "keyfile": "/path/to/apps.c1-ocp.myorg.com.key", "cafile": "/path/to/apps.c1-ocp.myorg.com.ca.crt"}
+----
+
+For example:
+
+----
+openshift_hosted_router_certificate={"certfile": "/home/cloud-user/star-apps.148.251.233.173.nip.io.cert.pem", "keyfile": "/home/cloud-user/star-apps.148.251.233.173.nip.io.key.pem", "cafile": "/home/cloud-user/ca-chain.cert.pem"}
+----
+
+Where the parameter values are: 
+
+* *certfile* contains the {product-title} router certificate.
+
+* *keyfile* contains the {product-title} wildcard key
+
+* *cafile* contains the root CA for this key and certificate. If an intermediate CA is in use it should contain both the intermediate and root CA.
+
+If these files are new to your {product-title} cluster, run an Ansible playbook to add these files to the {product-title} configuration files. 
+The playbook adds the certificate files to the *_/etc/origin/master/named_certificates/_* directory.
+
+----
+ansible-playbook /usr/share/ansible/openshift-ansible/playbooks/byo/openshift-cluster/config.yml
+----
+
+If the xref:../admin_solutions/install_config/redeploying_certificates.adoc#redeploying-all-certificates-current-ca[certificates expire 
+or you want to change the certificates] for some reason, run:
+
+----
+ansible-playbook /usr/share/ansible/openshift-ansible/playbooks/byo/openshift-cluster/redeploy-certificates.yml
+----
+
+[NOTE] 
+For the playbook to run, the certificate names must not changed. If the certificate names change, rerun the Ansible `config.yaml` playbook.
+
+[[configuring-custom-certificates-registry]]
+=== Using a Custom Certificate with the Image Registry
+
+The {product-title} image registry is an internal service that facilitates builds and deployments. Most of the communication with the registry is handled by internal components in {product-title}. As such, there should be no need to replace the certificate used by the registry service itself. 
+
+However, by default, the registry uses routes to allow external systems and users to do pulls and pushes of images. You can use a _re-encrypt route_ with a custom certificate that will be presented to external users instead of using the internal, self-signed certificate. 
+
+To configure this, add the following lines *_/etc/ansible/hosts_* file in the *[OSEv3:vars]*
+section. pointing to the certificates you would like used for the registry route.
+
+----
+openshift_hosted_registry_routehost=registry.apps.c1-ocp.myorg.com <1>
```

Comment:
Better to add an env var `openshift_hosted_registry_routetermination` here

Reply:
@ikke-t @sdodson Any thoughts on using `openshift_hosted_registry_routetermination` in place of `openshift_hosted_registry_routehost`? If so, `passthrough`or  'reencrypt`?

Reply:
I have no experience on it, I can't help